### PR TITLE
Update for Red 3.1

### PR DESCRIPTION
**Pull request by mikeshardmind**
This has the potential to have many broken things included here, especially given some abuses of config's internals which have been removed due to changes in config making them no longer safe for use.

Also should not be merged prior to the 3.1 release.

Have a lot of work to do to verify all of the changes here work as intended, but in theory, it should be ready to go.

### CODE REVIEW

File: embedmaker/embedmaker.py [hunk 1]
```diff
@@ -251,34 +250,6 @@ def __init__(self, bot):
             await group.owner.set(ctx.author.id)
             await group.embed.set(serialize_embed(e))
 
-    @_embed.command(name="list")
```
Comment (DiscordLiz) [inline embedmaker/embedmaker.py]: Removing this command intentionally?

File: rolemanagement/core.py [hunk 2]
```diff
@@ -487,153 +457,3 @@ def __init__(self, bot):
...
-                                        "REACTROLE", mid, em
-                                    ).channelid.set(channel.id)
-                                    break
-                    else:
-                        if not non_forbidden_encountered:
-                            await self.config.custom("REACTROLE", mid).clear()
-
-    # Stuff for clean interaction with react role entries
-
-    async def build_messages_for_react_roles(
```
Comment (DiscordLiz) [inline rolemanagement/core.py]: Isn't this needed?

File: rolemanagement/core.py [hunk 3]
```diff
@@ -487,153 +457,3 @@ def __init__(self, bot):
...
-                if emoji_info.isdigit():
-                    emoji = discord.utils.get(self.bot.emojis, id=int(emoji_info))
-                    emoji = emoji or f"A custom enoji with id {emoji_info}"
-                else:
-                    emoji = emoji_info
-
-                react_m = f"{role.name} is bound to {emoji} on {link}"
-                yield react_m
-
-    async def get_react_role_entries(
```
Comment (DiscordLiz) [inline rolemanagement/core.py]: And this?

File: rolemanagement/core.py [hunk 4]
```diff
@@ -143,43 +150,6 @@ def __init__(self, bot):
         """
         pass
 
-    @rgroup.command(name="viewreactions")
-    async def rg_view_reactions(self, ctx: commands.Context):
```
Comment (DiscordLiz) [inline rolemanagement/core.py]: And this?

File: embedmaker/embedmaker.py (hunk 1 above)
Comment (mikeshardmind) [reply embedmaker/embedmaker.py]: semi-intentionally. I'll need to rewrite the command.

File: rolemanagement/core.py (hunk 4 above)
Comment (mikeshardmind) [reply rolemanagement/core.py]: same as prior

File: rolemanagement/core.py (hunk 3 above)
Comment (mikeshardmind) [reply rolemanagement/core.py]: same as prior

File: rolemanagement/core.py (hunk 2 above)
Comment (mikeshardmind) [reply rolemanagement/core.py]: same as prior

**Comment by DiscordLiz** [comment]
Since you've asked for my opinion on this, don't remove the commands. Take the time to rewrite the logic. This is still blocked by changes upstream anyway.

**Comment by mikeshardmind** [comment]
Whether this has new command implementations ready or not, this will be merged at 3.1

I hope it won't come down to those pieces not being ready in time, but I'd prefer what is ready to be usable than none at all.

**Pull request closed without merging**